app/controllers/student_lessons_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+class StudentLessonsController < ApplicationController
+  def show
+    @student = Student.find params[:id]
+    @lesson = Lesson.find params[:lesson_id]
+    @grades = @student.current_grades_for_lesson_including_ungraded_skills params[:lesson_id]
+    @absence = @lesson.absences.map(&:student_id).include? @student.id
+  end
+
+  def update
+    student = Student.find params[:id]
+    student.grade_lesson params[:lesson_id], generate_grades_from_params(student)
+
+    mark_student_absence student
+    notice_and_redirect I18n.t(:student_graded), lesson_student_path
+  end
+
+  private
+
+  def absence
+    params.require(:student).permit(:absences)[:absences]
+  end
+
+  def mark_student_absence(student)
+    lesson = Lesson.find params[:lesson_id]
+    if absence == '1'
```

Comment:
Hmm, active record is supposed to handle that for you. If you're not assigning this to a model param, then there is a helper that you can use directly.